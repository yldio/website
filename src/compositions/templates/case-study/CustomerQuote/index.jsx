import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import GatsbyImage from 'gatsby-image';

import { Copy } from 'components/Typography';
import PageSection from 'components/PageSection';

import { QuoteWrapper, Quote, Author, Position, Container } from './styled';

const CustomerQuote = ({
  image,
  imageAlt,
  quote,
  author,
  position,
  subDescription,
  clientAppImage,
  clientAppImageAlt
}) => (
  <Fragment>
    <PageSection wide>
      <GatsbyImage alt={imageAlt} {...image.childImageSharp} />
      <Container>
        <QuoteWrapper>
          <Quote>{quote}</Quote>
          <Author>{author}</Author>
          <Position>{position}</Position>
        </QuoteWrapper>
      </Container>
    </PageSection>
    <PageSection>
      <Copy newLines>{subDescription}</Copy>
    </PageSection>
    <PageSection wide>
      <GatsbyImage
        alt={clientAppImageAlt}
        {...clientAppImage.childImageSharp}
      />
    </PageSection>
  </Fragment>
);

CustomerQuote.propTypes = {
  image: PropTypes.shape({
    childImageSharp: PropTypes.shape(GatsbyImage.propTypes).isRequired
  }).isRequired,
  imageAlt: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  position: PropTypes.string.isRequired,
  subDescription: PropTypes.string.isRequired,
  clientAppImage: PropTypes.shape({
    childImageSharp: PropTypes.shape(GatsbyImage.propTypes).isRequired
  }).isRequired,
  clientAppImageAlt: PropTypes.string.isRequired
};

export default CustomerQuote;
