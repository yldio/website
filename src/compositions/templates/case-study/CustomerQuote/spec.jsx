import React from 'react';
import { shallow } from 'enzyme';

import CustomerQuote from './index';

test('CustomerQuote', () => {
  const props = {
    image: {
      childImageSharp: {
        src: 'image.jpg'
      }
    },
    imageAlt: 'Quote image',
    quote: 'YLD offered amazing service,',
    author: 'Casey Bisson',
    position: 'Senior Director of Product Management - Joyent',
    subDescription:
      'Some amazing support copy with go here to reenforce whatever section it is relevant too.',
    clientAppImage: {
      childImageSharp: {
        src: 'image.jpg'
      }
    },
    clientAppImageAlt: 'Client App Image'
  };
  const wrapper = shallow(<CustomerQuote {...props} />);

  expect(wrapper).toMatchSnapshot();
});
