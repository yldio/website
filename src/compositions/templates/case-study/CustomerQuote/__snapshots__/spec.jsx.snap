// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerQuote 1`] = `
<React.Fragment>
  <PageSection
    wide={true}
  >
    <Image
      Tag="div"
      alt="Quote image"
      fadeIn={true}
      src="image.jpg"
    />
    <Container>
      <styled__QuoteWrapper>
        <styled__Quote>
          YLD offered amazing service,
        </styled__Quote>
        <styled__Author>
          Casey Bisson
        </styled__Author>
        <styled__Position>
          Senior Director of Product Management - Joyent
        </styled__Position>
      </styled__QuoteWrapper>
    </Container>
  </PageSection>
  <PageSection>
    <Typography__base
      newLines={true}
    >
      Some amazing support copy with go here to reenforce whatever section it is relevant too.
    </Typography__base>
  </PageSection>
  <PageSection
    wide={true}
  >
    <Image
      Tag="div"
      alt="Client App Image"
      fadeIn={true}
      src="image.jpg"
    />
  </PageSection>
</React.Fragment>
`;
